exp/internal/controllers/machinepool_controller_noderef.go

```diff
@@ -110,11 +110,20 @@ func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, cluster *
 			clusterv1.OwnerKindAnnotation:        mp.Kind,
 			clusterv1.OwnerNameAnnotation:        mp.Name,
 		}
-		if annotations.AddAnnotations(node, desired) {
-			if err := patchHelper.Patch(ctx, node); err != nil {
-				log.V(2).Info("Failed patch node to set annotations", "err", err, "node name", node.Name)
-				return ctrl.Result{}, err
+		// Add annotations.
+		annotations.AddAnnotations(node, desired)
```

Comment:
Q: Would it make sense to bring up the topic of in-place propagation on the Machine Pool machines via an issue?
(I guess it depends on MachinePool Machines being implemented first)

But fine for me to add the unitialized taint to MachinePool notes as well.

Reply:
I will open an issue and link it.

Reply:
Opened and issue: https://github.com/kubernetes-sigs/cluster-api/issues/8258 and also pinged on the MachinePool Machine PR: https://github.com/kubernetes-sigs/cluster-api/pull/7938#issuecomment-1462479151